src/arwave.py

```diff
@@ -1,122 +1,214 @@
+        min_wave = max(np.amin(arx_sky.wavelength.value), np.amin(obj_sky.wavelength.value))
+        max_wave = min(np.amax(arx_sky.wavelength.value), np.amax(obj_sky.wavelength.value))
+
+        #Smooth higher resolution spectrum by smooth_sig (flux is conserved!)
+        if np.median(obj_res) >= np.median(arx_res):
+            msgs.warn("New Sky has higher resolution than Archive.  Not smoothing")
+            #obj_sky_newflux = ndimage.gaussian_filter(obj_sky.flux, smooth_sig)
+        else:
+            #tmp = ndimage.gaussian_filter(arx_sky.flux, smooth_sig)
+            arx_sky = arx_sky.gauss_smooth(smooth_sig_pix*2.355)
+            #arx_sky.flux = ndimage.gaussian_filter(arx_sky.flux, smooth_sig)
+
+        # Define wavelengths of overlapping spectra
+        keep_idx = np.where((obj_sky.wavelength.value>=min_wave) &
+                             (obj_sky.wavelength.value<=max_wave))[0]
+        #keep_wave = [i for i in obj_sky.wavelength.value if i>=min_wave if i<=max_wave]
+
+        #Rebin both spectra onto overlapped wavelength range
+        if len(keep_idx) <= 50:
+            msgs.error("Not enough overlap between sky spectra")
+
+        else: #rebin onto object ALWAYS
+            keep_wave = obj_sky.wavelength[keep_idx]
+            arx_sky = arx_sky.rebin(keep_wave)
+            obj_sky = obj_sky.rebin(keep_wave)
+
+
+        #deal with bad pixels
+        msgs.work("Need to mask bad pixels")
+
+        #deal with underlying continuum
+        msgs.work("Need to deal with underlying continuum")
+
+        #Cross correlation of spectra
+        corr = np.correlate(arx_sky.flux, obj_sky.flux, "same")
+
+        #Create array around the max of the correlation function for fitting for subpixel max
+        # Restrict to pixels within max_shift of zero lag
+        lag0 = corr.size/2
+        mxshft = slf._argflag['reduce']['flexure']['max_shift']
+        max_corr = np.argmax(corr[lag0-mxshft:lag0+mxshft]) + lag0-mxshft
+        subpix_grid = np.linspace(max_corr-3., max_corr+3., 7.)
+
+        #Fit a 2-degree polynomial to peak of correlation function
+        fit = np.polynomial.polynomial.polyfit(subpix_grid, corr[subpix_grid.astype(np.int)], 2)
+        roots = np.roots([fit[2], fit[1], fit[0]])
+        max_fit = (roots[0]+roots[1])/2.
+        #   fit, other = arutils.gauss_lsqfit(subpix_grid, corr[subpix_grid.astype(np.int)], max_corr)
+
+        #Calculate and apply shift in wavelength
+        shift = float(max_fit)-lag0
+        msgs.info("Flexure correction of {:g} pixels".format(shift))
+        #model = (fit[2]*(subpix_grid**2.))+(fit[1]*subpix_grid)+fit[0]
+
+        if msgs._debug['flexure']:
+            debugger.xplot(arx_sky.wavelength, arx_sky.flux, xtwo=obj_sky.wavelength, ytwo=obj_sky.flux)
+            #debugger.xplot(arx_sky.wavelength, arx_sky.flux, xtwo=np.roll(obj_sky.wavelength.value,9), ytwo=obj_sky.flux*100)
+            debugger.set_trace()
+
+        # Simple interpolation
```

Comment:
Interesting suggestion.
But, I think the 'best' approach would probably 
be to add a higher order term to improve the flexure correction.

In essence, we are treating the sky spectrum as the 'truth'.